compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/ComponentDetectorCheck.kt

```diff
@@ -0,0 +1,41 @@
+package com.squareup.anvil.compiler.codegen.dagger
+
+import com.squareup.anvil.compiler.AnvilCompilationException
+import com.squareup.anvil.compiler.codegen.CodeGenerator
+import com.squareup.anvil.compiler.codegen.CodeGenerator.GeneratedFile
+import com.squareup.anvil.compiler.codegen.classesAndInnerClasses
+import com.squareup.anvil.compiler.codegen.hasAnnotation
+import com.squareup.anvil.compiler.daggerComponentFqName
+import com.squareup.anvil.compiler.daggerSubcomponentFqName
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
+import org.jetbrains.kotlin.psi.KtFile
+import java.io.File
+
+internal class ComponentDetectorCheck : CodeGenerator {
+  override fun generateCode(
+    codeGenDir: File,
+    module: ModuleDescriptor,
+    projectFiles: Collection<KtFile>
+  ): Collection<GeneratedFile> {
+    val component = projectFiles
+        .asSequence()
+        .flatMap { it.classesAndInnerClasses() }
+        .firstOrNull() { clazz ->
+          clazz.hasAnnotation(daggerComponentFqName) || clazz.hasAnnotation(
+              daggerSubcomponentFqName
+          )
+        }
+
+    if (component != null) {
+      throw AnvilCompilationException(
+          message = "Anvil cannot generate the code for Dagger components or subcomponents. In " +
+              "these cases the Dagger annotation processor is required. Enabling the Dagger " +
+              "annotation processor and turning on Anvil to generate Dagger factories is " +
+              "redundant. Set 'generateDaggerFactories' to false.",
```

Comment:
This check helped me to find the modules in our codebase that cannot make use of this feature.

Reply:
What about modules(gradle modules) which declare only `Subcomponent`. Dagger does not generate anything(maybe I'm wrong) for them, so why prohibit subcomponents?

Reply:
Actually I missed that. You're right. I filed #74.